posts/2021-04-16-mp-context-propagration-21004.adoc

```diff
@@ -0,0 +1,353 @@
+----
+
+For more information on `LdapRegistry`, see the link:https://openliberty.io/docs/latest/reference/feature/ldapRegistry-3.0.html[LDAP User Registry documentation].
+
+To enable this new beta function in your app, pull the All Beta Features package and add the LDAP User Registry 3.0 feature to your `server.xml` file:
+
+[source, xml]
+----
+<featureManager>
+  <feature>ldapRegistry-3.0</feature>
+</featureManager>
+----
+
+[#expandLocation]
+=== expandLocation property for AutoExpand
+
+With this enhancement users are now able to specify a expansion location (`expandLocation`) on the `applicationManager` configuration to be utilized when the `autoExpand` attribute is set to "true". As currently implemented, when an application is autoExpanded the default location for the expanded files are hard coded to `${server.config.dir}/apps/expanded/`.
+
+Now with this enhancement in place, the users are able to configure that location to a new value on the `filesystem`. Thus for example:
+
+`<applicationManager autoExpand="true" expandLocation="${server.config.dir}/myApps/" />` would result in the application being expanded at `${server.config.dir}/myApps/{appname}/`.
+
+This enhancement gives users more flexibility regarding the location of their expanded applications.
+
+Find out more at link:{url-prefix}/docs/latest/reference/config/applicationManager.html[Open Liberty Application Manager Documentation]
+
+[#requestTiming]
+== Ability to control whether thread dumps are collected when a hung request is detected in the Request Timing feature
+
+The Request Timing feature (`requestTiming-1.0`) provides diagnostic information when the duration of any request exceeds the configured threshold. It provides a way to monitor requests with respect to time. The feature can automatically detect slow and hung requests and provide detailed diagnostic information; warning messages, thread stacks, and the creation of thread dumps.
+
+When a hung request is detected in the Request Timing feature, a warning message is written in the messages log file along with a dump of the events that happened during the request. Following that, a set of three thread dumps will be initiated, 1 minute apart. After the completion of the three thread dumps, further set of three thread dumps are created only if new requests are detected to be hanging.
+
+Some operations teams do not want so many thread dumps to be generated due to performance overhead on requests that are known to be long. In previous Open Liberty releases, there was no option to disable the thread dumps from being generated.
+
+You can now control whether the Request Timing feature collects thread dumps. By setting the **new** `enableThreadDumps` Request Timing server configuration attribute to false, thread dumps will not be created during hung requests. If the new server configuration attribute is set to true or not specified at all, thread dumps will still be created.
```

Comment:
In 21.0.0.4, you can now control whether the Request Timing feature collects thread dumps. By setting the new `enableThreadDumps` Request Timing server configuration attribute to false, thread dumps will not be created during hung requests.  If `enableThreadDumps` is set to true or not specified at all, thread dumps will still be created.

Reply:
Done